it/uk/gov/hmrc/vatregisteredcompanies/repository/VatRegisteredCompaniesRepositoryISpec.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.vatregisteredcompanies.repository
+
+import uk.gov.hmrc.vatregisteredcompanies.helpers.IntegrationSpecBase
+import uk.gov.hmrc.vatregisteredcompanies.helpers.TestData._
+import uk.gov.hmrc.vatregisteredcompanies.repositories.PayloadWrapper
+
+class VatRegisteredCompaniesRepositoryISpec extends IntegrationSpecBase {
+  override def beforeEach(): Unit = {
+    deleteAll
+  }
+
+  "Method: deleteAll" when {
+    "there are no records in the database" should {
+      "Have no records" in {
+        totalCount shouldBe 0
+        val res = vatRegisteredCompaniesRepository.deleteAll()
+
+        whenReady(res) {result =>
+          result shouldBe ((): Unit)
+          totalCount shouldBe 0
+        }
+      }
+    }
+
+    "there is 1 record in the database" should {
+      "return unit and delete the record" in {
+        insertOne(acmeTradingWithVatNo1)
+        totalCount shouldBe 1
+        val res = vatRegisteredCompaniesRepository.deleteAll()
+
+        whenReady(res) {result =>
+          result shouldBe ((): Unit)
+          totalCount shouldBe 0
+        }
+      }
+    }
+
+    "there are multiple records in the database" should {
+      "return unit and delete all records" in {
+        insertMany(List(acmeTradingWithVatNo1, acmeTradingWithVatNo2, acmeTradingWithVatNo3))
+        totalCount shouldBe 3
+        val res = vatRegisteredCompaniesRepository.deleteAll()
+
+        whenReady(res) {result =>
+          result shouldBe ((): Unit)
+          totalCount shouldBe 0
+        }
+      }
+    }
+  }
+
+  "Method: process" when {
+    "the payload contains only create and updates" should {
```

Comment:
The payload looks like
case class Payload(
  createsAndUpdates: List[VatRegisteredCompany],
  deletes: List[VatNumber]
), in here it's only considered that payload have only create and updates, Should we test if only deletes are present and a case when both are present?

Reply:
I'm not sure where this comment is as its not on my branch.

Reply:
we can have a call about this